src/chainstate/stacks/db/transactions.rs

```diff
@@ -362,6 +362,14 @@ impl StacksChainState {
                     return Err(Error::InvalidStacksTransaction(msg));
                 }
 
+                // may not be sponsored
```

Comment:
Why are sponsored token transfers disallowed?

Reply:
It's forbidden by SIP 005 (see the description [here](https://github.com/blockstack/blockstack-core/blob/master/sip/sip-005-blocks-and-transactions.md#type-0-transferring-an-asset)).  The rationale is that it's non-sensical -- if you have STX already, you ought to be able to pay the transaction fee.

Reply:
Happy to re-enable it though.  Keeping it around certainly doesn't _hurt_ anything.

Reply:
Okay, that works for me. I'm fine with either keeping it around or leaving it forbidden.

Reply:
I reverted it and updated SIP 005 to allow sponsored token transfers.